Weather Alert/CItyTableViewController.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  CItyTableViewController.swift
+//  Weather Alert
+//
+//  Created by Jack Hopkins on 22/11/2016.
+//  Copyright © 2016 UKFast. All rights reserved.
+//
+
+import UIKit
+import CoreData
+
+class CityTableViewController: UITableViewController {
+    
+    // MARK: Variables
+        
+    var dataController: DataController { return AppShared.instances.dataController }
+    var cities: [City] {
+        let citiesFetch = NSFetchRequest<City>(entityName: "City")
+        do { return try dataController.managedObjectContext.fetch(citiesFetch) }
+        catch { fatalError("Failed to fetch employees: \(error)") }
+    }
+    
+    // MARK: UISearchController
+    
+    let searchController = UISearchController(searchResultsController: nil)
+    var filterCities: [City] = []
+    
+    
+    // MARK: Functions
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchController.searchResultsUpdater = self as UISearchResultsUpdating
+        searchController.dimsBackgroundDuringPresentation = false
+        definesPresentationContext = true
+        tableView.tableHeaderView = searchController.searchBar
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidLoad()
+        tableView.reloadData()
+    }
+    
+    @IBAction func searchEditingChanged(_ sender: UITextField) {
+        dataController.store()
+        self.tableView.reloadData()
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+}
+
+    extension CityTableViewController {
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        if let test = sender as? CityTableViewCell {
+            if let destinationVC = segue.destination as? CityDetailsViewController {
+                destinationVC.city = test.city
+            }
+        }
+    }
+}
+
+extension CityTableViewController {
+    
+    // MARK: Table View
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return (searchIsActive ? filterCities : cities ).count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cellIdentifier = "CityTableViewCell"
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath) as! CityTableViewCell
+        let city = (searchIsActive ? filterCities : cities )[indexPath.row]
+        
+        cell.city = city
```

Comment:
Although I did more than once, I would suggest not to mix model and view.
There should be some mediator that connects the two. Don't get used to anti-patterns.
If you use this function to populate the labels, in `prepareForSegue` you can get the indexpath of the cell that sends the event and extract the correct city using the indexpath's row variable.
Another option is to have a view model class that binds the model and the cell and populate the later.
In this app I would suggest using the first one as it's fairly a minor change and an easier one. If the project was more complex, the second one would have been better.